src/dataflow-types/src/client.rs

```diff
@@ -137,6 +142,11 @@ pub struct Peek<T = mz_repr::Timestamp> {
     /// If `Some`, the peek is only handled by the given replica.
     /// If `None`, the peek is handled by all replicas.
     pub target_replica: Option<ReplicaId>,
+    /// An `OpenTelemetryContext` to forward trace information along
+    /// to the compute-side to allow associating traces between
+    /// coord and compute.
```

Comment:
This code will soon be compute local, and so ideally the docs will only talk about compute stuff!

```suggestion
    /// An `OpenTelemetryContext` to forward trace information along
    /// to the compute worker to allow associating traces between
    /// the compute controller and the compute worker.
```